geonode/base/templatetags/base_tags.py

```diff
@@ -389,11 +389,20 @@ def get_visibile_resources(user):
 @register.simple_tag
 def display_edit_request_button(resource, user, perms):
     def _has_owner_his_permissions():
-        _owner_set = set(resource.BASE_PERMISSIONS.get('owner') +
-                         resource.BASE_PERMISSIONS.get('read') +
-                         resource.BASE_PERMISSIONS.get('write') +
-                         resource.BASE_PERMISSIONS.get('download')) - \
-            set(perms)
+        _owner_perms = set(
+            resource.BASE_PERMISSIONS.get('owner') +
+            resource.BASE_PERMISSIONS.get('read') +
+            resource.BASE_PERMISSIONS.get('write')
+        )
+
+        if resource.polymorphic_ctype.model in ['layer', 'document']:
```

Comment:
Uhm, the `polymorphic_ctype.model` check should be `dataset` here.

I would use the `resource.resource_type` property on master though.